heron/executor/src/python/heron_executor.py

```diff
@@ -200,8 +248,19 @@ def initialize(self):
     constructor
     """
     create_folders = 'mkdir -p %s' % self.log_dir
-    chmod_binaries = 'chmod a+rx . && chmod a+x %s && chmod +x %s %s %s' \
-        % (self.log_dir, self.tmaster_binary, self.stmgr_binary, self.heron_shell_binary)
+    chmod_x_binaries = [self.tmaster_binary, self.stmgr_binary, self.heron_shell_binary]
+    binaries_to_chmod = []
+
+    for binary in chmod_x_binaries:
+      stat_result = os.stat(binary)[stat.ST_MODE]
+      if not stat_result & stat.S_IXOTH:
+        binaries_to_chmod.append(binary)
```

Comment:
stat seems like a good approach, provided it works on all distros.

There's no need to do all chmods in one concatenated command, so better to break it up to make the code more readable. No need to build up arrays and concated strings, we can just do this:

```
- chmod log dir

- for each binary
  - if chmod needed
    - do chmod 
```